### PR TITLE
Automatically fallback to the core serializer if available

### PR DESCRIPTION
**Pull request by lsmith77**
follow up for https://github.com/FriendsOfSymfony/FOSRestBundle/pull/875

### CODE REVIEW

File: Resources/doc/1-setting_up_the_bundle.rst [hunk 1]
```diff
@@ -21,8 +21,14 @@ in the `installation chapter`_ of the Composer documentation.
...
-    But in this case, you need to manually set it up and configure FOSRestBundle
-    to use it via the ``service`` section in the app config
+    In this case, you need to `enable the core Serializer`_ in the FrameworkBundle.
+
+    Regardless if you use the JMSSerializerBundle or the FrameworkBundle to configure
+    the serializer service, the FOSRestBundle will automatically detect it, while giving
+    preference to using the JMSSerializer if both are enabled.
+
+    Alternatively manually set a serializer service and configure FOSRestBundle it via the
+    ``service`` section in the app config it the name of the service is not ``serializer``.
```
Comment (WouterJ) [inline Resources/doc/1-setting_up_the_bundle.rst]: I would put this in a new section:

````rst
C) Enable a Serializer
----------------------

This bundle needs a serializer to work correctly. In most cases,
you'll need to enable a serializer or install one. This bundle tries
the following this (in this order) to determine the serializer to use:

#. The one you configured using ``fos_rest.services.serializer`` (if you did).
#. The JMS serializer, if the `JMSSerializerBundle`_ is available (and registered).
#. The `Symfony Serializer`_ if it's enabled.
````

File: Resources/doc/1-setting_up_the_bundle.rst [hunk 2]
```diff
@@ -41,14 +31,26 @@ file of your project:
...
             // ...
         }
     }
 
+C) Enable a Serializer
+----------------------
+
+This bundle needs a serializer to work correctly. In most cases,
+you'll need to enable a serializer or install one. This bundle tries
+the following this (in this order) to determine the serializer to use:
```
Comment (merk) [inline Resources/doc/1-setting_up_the_bundle.rst]: Sentence is a bit weird..

`..tries the following (in order) to determine...`

File: Resources/doc/1-setting_up_the_bundle.rst [hunk 3]
```diff
@@ -41,14 +31,26 @@ file of your project:
...
+#. The JMS serializer, if the `JMSSerializerBundle`_ is available (and registered).
+#. The `Symfony Serializer`_ if it's enabled (or any service called ``serializer``).
+
 That was it!
 
 .. _`installation chapter`: https://getcomposer.org/doc/00-intro.md
 .. _`JMSSerializer`: https://github.com/schmittjoh/serializer
 .. _`JMSSerializerBundle`: https://github.com/schmittjoh/JMSSerializerBundle
 .. _`Symfony Serializer`: https://github.com/symfony/Serializer
+.. _`enable the core Serializer`: http://symfony.com/doc/current/cookbook/serializer.html
```
Comment (merk) [inline Resources/doc/1-setting_up_the_bundle.rst]: This doesnt appear to be used any more?

Comment (lsmith77) [inline Resources/doc/1-setting_up_the_bundle.rst]: see https://github.com/FriendsOfSymfony/FOSRestBundle/commit/2fb6d7018ba30933010a80f29c3673b2bb0ef6e3

File: Resources/doc/1-setting_up_the_bundle.rst (hunk 2 above)
Comment (lsmith77) [inline Resources/doc/1-setting_up_the_bundle.rst]: see https://github.com/FriendsOfSymfony/FOSRestBundle/commit/2fb6d7018ba30933010a80f29c3673b2bb0ef6e3

**Pull request merged**